web/src/app/lists/FlatList.tsx

```diff
@@ -154,23 +133,58 @@ export default function FlatList({
   items,
   inset,
   transition,
+  toggleDnD,
   ...listProps
 }: FlatListProps): JSX.Element {
   const classes = useStyles()
 
+  // drag and drop stuff
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    }),
+  )
+  const [dndItems, setDndItems] = useState(
+    // use IDs to sort, fallback to index
+    items.map((i, idx) => (i.id ? i.id : idx.toString())),
+  )
```

Comment:
```suggestion
  const [dndItems, setDndItems] = useState(
    // use IDs to sort, fallback to index
    items.map((i, idx) => (i.id ? i.id : idx.toString())),
  )
  
  useEffect(() => {
    setDndItems(items.map((i, idx) => (i.id ? i.id : idx.toString())))
  }, [items])
```

The `dndItems` can get out of sync with the list `items` if users are removed from the rotation, adding a useEffect to keep updating on change seems to fix it